lgc/builder/BuilderBase.cpp

```diff
@@ -61,8 +61,22 @@ CallInst *BuilderCommon::CreateNamedCall(StringRef funcName, Type *retTy, ArrayR
     func->setCallingConv(CallingConv::C);
     func->addFnAttr(Attribute::NoUnwind);
 
-    for (auto attrib : attribs)
-      func->addFnAttr(attrib);
+    for (auto attrib : attribs) {
+      switch (attrib) {
+      default:
+        func->addFnAttr(attrib);
+        break;
+      case Attribute::ReadNone:
+        func->setDoesNotAccessMemory();
+        break;
+      case Attribute::ReadOnly:
+        func->setOnlyReadsMemory();
+        break;
+      case Attribute::WriteOnly:
+        func->setOnlyWritesMemory();
+        break;
+      }
```

Comment:
One reason I wasn't sure about #2119 was that this function is just hiding the fact that ReadNone, ReadOnly and WriteOnly aren't supported as function attributes any more.
I wondered if an approach that added an extra parameter for memory effects might be the cleaner way to go? Not really sure though.

Reply:
Yes, I have been working on such a patch, but I don't think it can (easily) be done in a way that is backwards compatible with the current state of https://github.com/GPUOpen-Drivers/llvm-project/commits/amd-gfx-gpuopen-dev because all callers of CreateNamedCall would have to pass in a MemoryEffects object, but MemoryEffects does not even exist in that branch.

Reply:
Ok - makes sense.